libmscore/repeatlist.cpp

```diff
@@ -93,12 +75,74 @@ Measure* Score::searchLabelWithinSectionFirst(const QString& s, Measure* section
 RepeatSegment::RepeatSegment()
       {
       tick       = 0;
-      len        = 0;
       utick      = 0;
       utime      = 0.0;
       timeOffset = 0.0;
       }
 
+RepeatSegment::RepeatSegment(RepeatSegment * const rs, Measure * const fromMeasure, Measure * const untilMeasure)
+      {
+      tick       = 0;
+      utick      = 0;
+      utime      = 0.0;
+      timeOffset = 0.0;
+      //copy the measure list
+      auto it = rs->measureList.cbegin();
+      if (nullptr != fromMeasure) {
+            //skip until fromMeasure
+            while ((it->first != fromMeasure) && ((++it) != rs->measureList.cend()));
+            }
+      while (it != rs->measureList.cend()) {
+            if ((nullptr != untilMeasure) && (it->first->no() > untilMeasure->no())) {
+                  break;
+                  }
+            measureList.push_back(std::make_pair(it->first, it->second));
+            ++it;
+            }
+      if (!measureList.empty()) {
+            tick = measureList.cbegin()->first->tick();
+            }
+      }
+
+void RepeatSegment::addMeasure(Measure * const m)
+      {
+      if (measureList.empty()) {
+            tick = m->tick();
+            }
+      measureList.push_back(std::make_pair(m, m->playbackCount()));
+      }
+
+bool RepeatSegment::containsMeasure(Measure const * const m) const
```

Comment:
This is up to `std::any_of()`method. No need to create wheels.
`return std::any_of(measureList.begin(), measureList.end(), [&m](const auto& elem) { return elem.first == m; });`

Reply:
We don't use lambda nor std::any_of() in the rest of the code.